src/lua/fiber.c

```diff
@@ -653,6 +658,14 @@ lbox_fiber_storage(struct lua_State *L)
 	struct fiber *f = lbox_checkfiber(L, 1);
 	int storage_ref = f->storage.lua.storage_ref;
 	if (storage_ref == FIBER_LUA_NOREF) {
+		if (f->flags & FIBER_IS_DEAD) {
+			diag_set(IllegalParams, "Fiber is dead");
```

Comment:
In other places the error message is a bit different: "the fiber is dead". Let's be consistent with that:

https://github.com/tarantool/tarantool/blob/319357d5973d15d08b8eda6a230eada08b710802/src/lua/fiber.c#L592

Also, what about adding a helper function like `lbox_checkfiber` that would raise this error if the fiber exists but is dead?

Reply:
Discussed f2f and decided to just change message to 'the fiber is dead'. In other cases the fiber does not exist.

Reply:
```diff
diff --git a/src/lua/fiber.c b/src/lua/fiber.c
index 585576b46..c8415649f 100644
--- a/src/lua/fiber.c
+++ b/src/lua/fiber.c
@@ -659,7 +659,7 @@ lbox_fiber_storage(struct lua_State *L)
        int storage_ref = f->storage.lua.storage_ref;
        if (storage_ref == FIBER_LUA_NOREF) {
                if (f->flags & FIBER_IS_DEAD) {
-                       diag_set(IllegalParams, "Fiber is dead");
+                       diag_set(IllegalParams, "the fiber is dead");
                        luaT_error(L);
                }
                struct trigger *on_stop = xmalloc(sizeof(*on_stop));
```